packages/relay/src/lib/services/debugService/index.ts

```diff
@@ -0,0 +1,219 @@
+          type: action.call_operation_type,
+          from: resolvedFrom,
+          to: resolvedTo,
+          gas: numberTo0x(action.gas),
+          gasUsed: numberTo0x(action.gas_used),
+          input: getContract?.bytecode ?? action.input,
+          output: getContract?.runtime_bytecode ?? action.result_data,
+          value: numberTo0x(action.value),
+        };
+      }),
+    );
+  }
+
+  /**
+   * Returns an address' evm equivalence.
+   *
+   * @async
+   * @param {string} address - The address to be resolved.
+   * @param {[string]} types - The possible types of the address.
+   * @param {string} requestIdPrefix - The request prefix id.
+   * @returns {Promise<string>} The address returned as an EVM address.
+   */
+  async resolveAddress(
+    address: string,
+    types = [constants.TYPE_CONTRACT, constants.TYPE_TOKEN, constants.TYPE_ACCOUNT],
+    requestIdPrefix?: string,
+  ): Promise<string> {
+    const entity = await this.mirrorNodeClient.resolveEntityType(
+      address,
+      types,
+      'debug_traceTransaction',
+      requestIdPrefix,
+    );
+
+    if (
+      entity &&
+      (entity.type === constants.TYPE_CONTRACT || entity.type === constants.TYPE_ACCOUNT) &&
+      entity.entity?.evm_address
+    ) {
+      return entity.entity.evm_address;
+    }
+
+    return address;
+  }
+
+  /**
+   * Returns the final formatted response for callTracer config.
+   *
+   * @async
+   * @param {string} transactionHash - The hash of the transaction to be debugged.
+   * @param {any} tracerConfig - The tracer config to be used.
+   * @param {string} requestIdPrefix - The request prefix id.
+   * @returns {Promise<object>} The formatted response.
+   */
+  async callTracer(transactionHash: string, tracerConfig: any, requestIdPrefix?: string): Promise<object> {
+    try {
+      const [actionsResponse, transactionsResponse] = await Promise.all([
+        this.mirrorNodeClient.getContractsResultsActions(transactionHash, requestIdPrefix),
+        this.mirrorNodeClient.getContractResultWithRetry(transactionHash),
+      ]);
+
+      const { call_type: type } = actionsResponse.actions[0];
+      const formattedActions = await this.formatActionsResult(actionsResponse.actions, requestIdPrefix);
+
+      const {
+        from,
+        to,
+        amount,
+        gas_limit: gas,
+        gas_used: gasUsed,
+        function_parameters: input,
+        call_result: output,
+        error_message: error,
+        result,
+      } = transactionsResponse;
+
+      const [fromToEvmAddress, toToEvmAddress] = await Promise.all([
+        this.resolveAddress(
+          from,
+          [constants.TYPE_CONTRACT, constants.TYPE_TOKEN, constants.TYPE_ACCOUNT],
+          requestIdPrefix,
+        ),
+        this.resolveAddress(
+          to,
+          [constants.TYPE_CONTRACT, constants.TYPE_TOKEN, constants.TYPE_ACCOUNT],
+          requestIdPrefix,
+        ),
+      ]);
+
+      const value = amount === 0 ? '0x0' : numberTo0x(amount);
```

Comment:
nit: Use constants for `0x0` and `SUCCESS`. I think these are already defined in places